File: examples/Fortran/openmp_converted/Loop_configuration_example_omp.f90
program loop_configuration
    use ISO_FORTRAN_ENV, only : INT32, REAL64
    use openacc
use omp_lib
    implicit none    
    integer(kind=INT32), parameter        :: n = 500
    integer(kind=INT32), dimension(n,n,n) :: table
    integer(kind=INT32)                   :: ngangs, nworkers, nvectors    
    integer(kind=INT32)                   :: i, j, k

    ngangs   = 450
    nworkers = 4
    nvectors = 16 

    !$acc parallel loop gang num_gangs(ngangs) num_workers(nworkers) vector_length(nvectors) copyout(table(:,:,:))
!$omp target teams loop num_teams(ngangs) thread_limit(nvectors) map(from:table(:,:,:))
    do k = 1, n
        !$acc loop worker
!$omp loop
        do j = 1, n
            !$acc loop vector
!$omp loop 
            do i = 1, n
                table(i,j,k) = i + j*1000 + k*1000*1000
            enddo
        enddo
    enddo

    print *, table(1,1,1), table(n,n,n)
end program loop_configuration      

! Code was translated using: intel-application-migration-tool-for-openacc-to-openmp Loop_configuration_example.f90

File: examples/C/openmp_converted/Loop_configuration_example_omp.c
#include <stdio.h>
#include <stdlib.h>
#include <openacc.h>
#include <omp.h>
int main(void)
{ 
    int n = 200;
    int ngangs = 1, nworkers = 2, nvectors = 32;
    size_t table[n*n*n];
    
#pragma acc parallel loop gang num_gangs(ngangs) num_workers(nworkers) vector_length(nvectors) copyout(table[0:n*n*n])
#pragma omp target teams loop map(from:table[0:n*n*n])
    for (int i=0; i<n; ++i)
    {
#pragma acc loop worker
#pragma omp loop
        for (int j=0; j<n; ++j)
        {
            #pragma acc loop vector
#pragma omp loop
            for (int k=0; k<n; ++k) table[i*n*n + j*n + k] = k + 1000*j + 1000*1000*i;
        }
    }
    printf("%d %d\n",table[0],table[n*n*n-1]);
}

// Code was translated using: /home/very/bin/acc2mp -async=nowait -specify-language=C -generate-report Loop_configuration_example.c
